app/src/main/java/model/objects/building/Building.java

```diff
@@ -1,11 +1,109 @@
 package model.objects.building;
 
+import controller.elevatorSystemController.ElevatorSystemSettings;
+import lombok.Getter;
 import model.objects.elevator.Elevator;
+import tools.Vector2D;
 
 import java.util.LinkedList;
+import java.util.Random;
 
 public class Building {
     public final LinkedList<Elevator> ELEVATORS = new LinkedList<>();
+    public final ElevatorSystemSettings SETTINGS;
+    @Getter
+    public double wallSize;
+
+    public Building(ElevatorSystemSettings settings) {
+        this.SETTINGS = settings;
+        for (int i = 0; i < SETTINGS.MAX_ELEVATORS_COUNT; i++) {
+            var newElevator = new Elevator(settings);
+            newElevator.setVisible(false);
+            ELEVATORS.add(newElevator);
+        }
+        updateElevatorsPosition();
+    }
+
+    public void updateElevatorsPosition() {
+        this.wallSize = ((double) SETTINGS.BUILDING_SIZE.y) / SETTINGS.FLOORS_COUNT;
+        double distanceBetweenElevators = ((double) SETTINGS.BUILDING_SIZE.x) / (SETTINGS.getElevatorsCount() + 1);
+        var elevatorIterator = ELEVATORS.iterator();
+        for (int i = 0; i < SETTINGS.MAX_ELEVATORS_COUNT; i++) {
+            var elevator = elevatorIterator.next();
+
+            if (i < SETTINGS.getElevatorsCount()) {
+                elevator.setWallSize(wallSize);
+                if (elevator.isVisible()) {
+                    elevator.setPosition(new Vector2D(distanceBetweenElevators * (i + 1), elevator.getPosition().y));
+                } else {
+                    elevator.setVisible(true);
```

Comment:
No nie , to jest dziwne. Mozemy dodac mozliwosc usuwania i dodawanie wind a nie chowanie ich ?

Reply:
kk